### PR TITLE
pacific: pybind/mgr: fixup after upgrading tox versions

### PR DESCRIPTION
**Pull request by adk3798**
We upgraded the tox version on the Jenkins machines and some changes are needed in order for make check to pass now.

Backport of https://github.com/ceph/ceph/pull/49322, https://github.com/ceph/ceph/pull/49321, partially https://github.com/ceph/ceph/pull/49359 (included some unnecessary things I've removed here and will revert in main later)

A lot more merge conflicts in this one than the quincy backport. Hopefully all handled correctly and make check still passes. 

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by adk3798** [comment]
```
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python3 (missing: Python3_LIBRARY Python3_INCLUDE_DIR
  Development) (found suitable exact version "3.8.10")
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/modules/FindPython/Support.cmake:1181 (find_package_handle_standard_args)
  cmake/modules/FindPython3.cmake:181 (include)
  CMakeLists.txt:485 (find_package)
```
in make check. A bit surprising. Don't think we're adding Python3 as a dep in this PR.

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
make check (specifically run-tox-qa test) failing now with
```
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/test_import.py", line 43, in <module>
    _import_file(p)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/test_import.py", line 22, in _import_file
    mod = mod_spec.loader.load_module()
  File "<frozen importlib._bootstrap_external>", line 548, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 1063, in load_module
  File "<frozen importlib._bootstrap_external>", line 888, in load_module
  File "<frozen importlib._bootstrap>", line 290, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 719, in _load
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/tasks/ceph_fuse.py", line 9, in <module>
    from tasks.cephfs.fuse_mount import FuseMount
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/tasks/cephfs/fuse_mount.py", line 13, in <module>
    from tasks.cephfs.mount import CephFSMount
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/tasks/cephfs/mount.py", line 12, in <module>
    from IPy import IP
ModuleNotFoundError: No module named 'IPy'
Importing tasks.ceph_fuse from tasks/ceph_fuse.py                               import-tasks: exit 1 (2.77 seconds) /home/jenkins-build/build/workspace/ceph-pull-requests/qa> python test_import.py 'tasks/**/*.py' pid=2364562
  flake8: OK (16.22=setup[11.89]+cmd[4.33] seconds)
  import-tasks: FAIL code 1 (103.85=setup[101.08]+cmd[2.77] seconds)
  evaluation failed :( (120.20 seconds)
```

File: qa/CMakeLists.txt [hunk 1]
```diff
@@ -5,5 +5,5 @@ endif()
 
 if(WITH_TESTS)
   include(AddCephTest)
-  add_tox_test(qa TOX_ENVS py3 flake8 import-tasks)
+  add_tox_test(qa TOX_ENVS flake8 import-tasks)
```
Comment (tchaikov) [inline qa/CMakeLists.txt:8]:
```suggestion
  add_tox_test(qa TOX_ENVS py3 flake8)
```

File: src/pybind/mgr/tox.ini [hunk 2]
```diff
@@ -1,7 +1,6 @@
 [tox]
 envlist =
     py3,
-    py37,
```
Comment (tchaikov) [inline src/pybind/mgr/tox.ini:4]: i don't think we need to backport this commit.

Comment (adk3798) [reply src/pybind/mgr/tox.ini:4]: should I drop it from the quincy backport as well?

**Review by ljflores — APPROVED** [review]

File: src/pybind/mgr/tox.ini (hunk 2 above)
Comment (adk3798) [reply src/pybind/mgr/tox.ini]: > should I drop it from the quincy backport as well?

no, we should not

**Pull request merged**